truffle/src/main/java/org/jruby/truffle/RubyEngine.java

```diff
@@ -35,6 +36,7 @@ public RubyEngine(RubyInstanceConfig instanceConfig) {
 
         engine = PolyglotEngine.newBuilder()
                 .config(RubyLanguage.MIME_TYPE, INSTANCE_CONFIG_KEY, instanceConfig)
+                .config(RubyLanguage.MIME_TYPE, OptionsCatalogue.ARGUMENTS.getName(), instanceConfig.getArgv())
```

Comment:
It's already a hash map - one that the polyglot engine maintains - these `config` calls are like `put`. And it's already private to our language.

There are only ten or so fields like this that we need to pass. We don't need to do this for options that `RubyInstanceConfig` doesn't currently set.